### ISSUE TITLE
Adding configuration page to open ports?

### ISSUE THREAD

**Issue by JoKeyser** [freedombox/Plinth#1110]
From the mailing list,
On 10/31/2017 07:02 PM, Pierre L. wrote:
> Hey,
> [Opening ports manually] can be a cool feature to add this in the GUI!
> Something like home ISP routers features... A column with the service
> name + ports (only one, few, or a range), TCP/UDP/etc...
> Then this GUI will add those ports to the firewall daemon config.
> All the default Freedombox ports will be listed (with the service name
> too ;) )and can't be used by the user.

**Comment by mray** [comment]
Tangential thoughts:
I think what is badly missing is a nice way to have an overview of what ports FreedomBox requires to be open, with indication whether they are or not. Diagnostics only give limited overview. Maybe a dedicated "Ports" module makes sense? Opening and closing could also be made accessible there probably-

Also each module that requires open ports needs to signal that in the GUI consistently, too. Currently you have to *investigate* what exactly you need to set up in your router.

**Comment by JosephKiranBabu** [comment]
Plinth generally opens the required ports for an application automatically.
This is an advanced feature that people with some system administration skills will want to use. So, I'd recommend delegating this to Cockpit. They already have an epic for this on their [trello board](https://trello.com/c/DVQIn8AC/65-firewall-support).

**Comment by JoKeyser** [comment]
Thanks for your comments. A *passive* overview of all opened ports for the purpose of setting up your home router (if the fbx is behind it) seems helpful. I also agree that the manual control to plinth is an advanced feature what we should delegate to Cockpit, to avoid feature creep - great that they'll include it.

**Comment by JosephKiranBabu** [comment]
Yes, I agree that information about the list of ports to be opened is useful for port-forwarding, since not all routers support DMZ. 
Maybe we can show this information in the `configure` page where the user sets the domain name.

**Comment by mray** [comment]
I like that idea.
A simple list that indicates the status would already be an improvement:


Appname | Required port | open/closed
-------------|-------------------|----------------

**Comment by JoKeyser** [comment]
I guess one would only have to add the actual number to the *Configuration -> Firewall* overview?

![fbx_firewall_tab](https://user-images.githubusercontent.com/8722846/32508520-e4d1267c-c3ea-11e7-95b3-e39f364e8484.png)

**Comment by JoKeyser** [comment]
@JosephKiranBabu Ah okay I see, your considerations make sense to me. I'm not sure about putting it directly on the general Configure page, as the ports are a very specific aspect. But surely there should be a link to wherever the ports are listed. Similarly, there should be a link in the Network configuration to that overview.
So where to put the overview table? I still think the Firewall page is quite natural, since it's also interesting to see what's currently enabled, for what service... so what about calling the page "Firewall and Ports", or something like that?

**Comment by mray** [comment]
If hardly anyone would make use of that, the top level "configuration" module isn't the right place. It should remain simple and relevant if possible.
